Fail to build on solaris 10

Hi，
I'm building log4cplus-2.x on solaris 10, but it's report a error with "src/lockfile.cxx":
``` c
libtool: compile:  /opt/csw/bin/g++ -DHAVE_CONFIG_H -I./include -I./threadpool -I./include -I./catch/single_include -DCATCH_CONFIG_PREFIX_ALL=1 -DNDEBUG -DINSIDE_LOG4CPLUS -D_GNU_SOURCE=1 -D_REENTRANT -m64 -fPIC -D_REENTRANT -Wno-unused-variable -Wno-unused-local-typedefs -Wno-unused-function -std=gnu++11 -Wall -fdiagnostics-show-caret -ftrack-macro-expansion -fdiagnostics-color=auto -Wextra -pedantic -Wstrict-aliasing -Wstrict-overflow -Woverloaded-virtual -Wold-style-cast -Wundef -Wshadow -Wformat -Wnoexcept -Wsuggest-attribute=format -Wsuggest-attribute=noreturn -Wno-variadic-macros -O2 -fvisibility=hidden -D_REENTRANT -MT src/liblog4cplus_la-lockfile.lo -MD -MP -MF src/.deps/liblog4cplus_la-lockfile.Tpo -c src/lockfile.cxx  -fPIC -DPIC -o src/.libs/liblog4cplus_la-lockfile.o
src/lockfile.cxx: In member function 'void log4cplus::helpers::LockFile::open(int) const':
src/lockfile.cxx:240:40: error: 'loglog' was not declared in this scope
     if (! trySetCloseOnExec (data->fd, loglog))

```

Should i declare loglog first on method LockFile::open() ?

--- a/src/lockfile.cxx
+++ b/src/lockfile.cxx
@@ -237,7 +237,7 @@ LockFile::open (int open_flags) const
             + lock_file_name, true);
 
 #if ! defined (O_CLOEXEC)
-    if (! trySetCloseOnExec (data->fd, loglog))
+    if (! trySetCloseOnExec (data->fd))
         getLogLog ().warn (
             tstring (LOG4CPLUS_TEXT("could not set FD_CLOEXEC on file "))
             + lock_file_name);

